src/phongo_bson_encode.c

```diff
@@ -466,6 +466,15 @@ static void php_phongo_zval_to_bson_internal(zval* data, php_phongo_field_path*
 			}
 
 			if (instanceof_function(Z_OBJCE_P(data), php_phongo_packedarray_ce)) {
+				/* If we are at the root-level, PackedArray instances should be
+				 * prohibited unless PHONGO_BSON_ALLOW_ROOT_ARRAY is set. */
+				bool is_root_level = (field_path->size == 0);
+
+				if (is_root_level && !(flags & PHONGO_BSON_ALLOW_ROOT_ARRAY)) {
+					phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE, "%s cannot be serialized as a root document", ZSTR_VAL(Z_OBJCE_P(data)->name));
```

Comment:
I used UnexpectedValueException here for consistency with other BSON encoding exceptions. That is also the only exception we document in [`MongoDB\BSON\fromPHP()`](https://www.php.net/manual/en/function.mongodb.bson-fromphp.php).

It does make the tests a bit odd, since those mostly expect InvalidArgumentException for option validation errors; however, this isn't an error most users should ever encounter anyway.

Lastly, I used "root document" here but left the original error messages for enums and Type instances as "root element". I don't think we need to change those, but I can if you like. Referring to a "document" here seemed more impactful since it contrasts with PackedArray.